### DISCUSSION TITLE
Adding LSP support for Ethereum Solidity

### DISCUSSION THREAD

**Discussion by AdamBD** [neoclide/coc.nvim#3019] [Extension request]
Hi guys,

Are there any plans to add solidity? Developing smart contracts in Coc.nvim would be great!

**Comment by Adoliin** [comment]
+1

**Comment by chemzqm** [comment]
Check if there's language server for that.

**Reply by pxue** [reply]
[nvim-lsp](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#solidity_ls) supports solidity-ls via [solidity-language-server](https://www.npmjs.com/package/solidity-language-server)

**Comment by qiuxiang** [comment]
I am writing a coc plugin for solidity: https://github.com/qiuxiang/coc-solidity

**Comment by alannotnerd** [comment]
Hi, I just port the lsp from hardhat-vscode. Pls check that. https://github.com/alannotnerd/solsp
```
  "languageserver": {
    "solidity": {
      "command": "solsp",
      "args": ["--stdio"],
      "rootPatterns": [".git/"],
      "trace.server": "verbose",
      "filetypes": ["solidity"]
    }
  }

```
